### PR TITLE
client: add FetchServerAddr for host fallback on session failure

### PR DESCRIPTION
**Pull request by rjeczalik**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -253,10 +253,12 @@ func (s *Server) controlHandler(w http.ResponseWriter, r *http.Request) (ctErr e
 
 	conn, _, err := hj.Hijack()
 	if err != nil {
-		return fmt.Errorf("hijack not possible %s", err)
+		return errors.New("hijack not possible: " + err.Error())
```
Comment (fatih) [inline server.go]: Any reason you changed this? I think the previous one was more explicit :)

**Comment by fatih** [comment]
LGTM :+1:

File: server.go (hunk 1 above)
Comment (rjeczalik) [inline server.go]: No reason, changed back. :-)

**Comment by cihangir** [comment]
:ok_hand:

**Pull request merged**